src/Grpc.AspNetCore.Server/GrpcBindingOptions.cs

```diff
@@ -31,9 +32,7 @@ public class GrpcBindingOptions<TService>
         /// </summary>
         public Action<ServiceBinderBase, TService> BindAction { get; set; }
```

Comment:
What’s this for? When would you override it?

Reply:
gRPC will code gen a method that we use to discover methods on the service:

```cs
    /// <summary>Register service method with a service binder with or without implementation. Useful when customizing the  service binding logic.
    /// Note: this method is part of an experimental API that can change or be removed without any prior notice.</summary>
    /// <param name="serviceBinder">Service methods will be bound by calling <c>AddMethod</c> on this object.</param>
    /// <param name="serviceImpl">An object implementing the server-side handling logic.</param>
    public static void BindService(grpc::ServiceBinderBase serviceBinder, GreeterBase serviceImpl)
    {
      serviceBinder.AddMethod(__Method_SayHello, serviceImpl == null ? null : new grpc::UnaryServerMethod<global::Greet.HelloRequest, global::Greet.HelloReply>(serviceImpl.SayHello));
      serviceBinder.AddMethod(__Method_SayHellos, serviceImpl == null ? null : new grpc::ServerStreamingServerMethod<global::Greet.HelloRequest, global::Greet.HelloReply>(serviceImpl.SayHellos));
    }
```

Gprc.Tools always places this in the same location, but if another tool is used to generate the services - perhaps you want to use something other than Google.Protobuf for serialization - then the method could be in a different location. This delegate allows you to specify the method explicitly.

```cs
    builder.MapGrpcService<GreeterService>(options => options.BindAction = Greet.Greeter.BindService);
```

Reply:
Oh no I’m aware I’m mostly asking if we think this is useful to expose in this first class way. Do we think people should be doing this? Is this something we’re going to generate in the future?

Reply:
I see explicitly setting the bind action as an escape hatch for when  `MapGrpcService<T>()` doesn't work. If we don't provide an escape hatch then any code gen-ed types that don't have the right structure (and the structure is pretty weird tbh) will be locked out from being used with `Grpc.AspNetCore`.